### ISSUE TITLE
Make More TX Log Availabe Through RPC

### ISSUE THREAD

**Issue by brianlong** [solana-labs/solana#22301]
#### Problem

The size of the TX log data returned through RPC is limited. Many of the applications write extensive logs and the log size exceeds the RPC limit. To properly analyze compute units per leader, we need to see all of the logs for a transaction.

#### Proposed Solution

Allow RPC operators to indicate the amount of TX log data to return. e.g. `--rpc-log-data-limit 100`

**Comment by CriesofCarrots** [comment]
Afaik, the only limit on log sizes is the limit on how much log data is stored by a node (with `--enable-rpc-transaction-history` and `--enable-cpi-and-log-storage` params); rpc endponts like getTransaction and getBlock return all the logs stored, without limit. Just making sure we're on the same page; is the log-data storage limit the one you'd like to be able to configure?
The current cap is ~10kb of logs per transaction: https://github.com/solana-labs/solana/blob/69e632a3372b2d93e89095ff8fef718e98616ac2/program-runtime/src/log_collector.rs#L4

**Comment by brianlong** [comment]
Yes, that's it. 10K is too small and we are missing log entries for some TXs. It will be nice to override the default value with a CLI parameter.

**Comment by willhickey** [comment]
Fixed by #25914 and will be included in version v1.11.3
